sdk/src/message.rs

```diff
@@ -127,6 +127,7 @@ fn get_program_ids(instructions: &[Instruction]) -> Vec<Pubkey> {
 pub struct MessageHeader {
     /// The number of signatures required for this message to be considered valid. The
     /// signatures must match the first `num_required_signatures` of `account_keys`.
+    /// NOTE: Serialization-related changes must be paired with the direct read at sigverify.
```

Comment:
As bonuses, I added precautionary comments to prevent this bug from reoccurring in the future...

Reply:
Can you also add a test that fails if the serialization changes?

Reply:
Well, I could do that of course, however I think that will be out of scope for this PR, assuming you're meaning to write some robust set of unit tests for each plausible outcome of `get_packet_offsets`. This PR getting relatively large as one from an outside contributor already.

I've already wrote units tests, directly spotting the changed behavior: [1](https://github.com/solana-labs/solana/pull/6236/files#diff-ad695894f5a727f4ce60e68987c4e35aR607), [2](https://github.com/solana-labs/solana/pull/6236/files#diff-ad695894f5a727f4ce60e68987c4e35aR469) and [3](https://github.com/solana-labs/solana/pull/6236/files#diff-ad695894f5a727f4ce60e68987c4e35aR441).

I understand those yet-to-be-written unit tests will be invaluable, considering sigverify will directly be faced to the outside world (= the public Internet for the permission-less DLT like solana). But the concern already are covered by the issues like #5414 and #6339.

Anyway, if there are some thoughts I'm missing or better unit test scenario, please tell me!

Reply:
Ok, no problem, we can take on writing that test.

Reply:
Thanks for consideration! Hopefully, writing it will be me if I find some free time!